nft_ingester/src/index_syncronizer.rs

```diff
@@ -771,9 +1100,29 @@ mod tests {
             false,
         );
         let (_, rx) = tokio::sync::broadcast::channel::<()>(1);
-        synchronizer
-            .synchronize_asset_indexes(&rx, 0)
-            .await
-            .unwrap();
+        let synchronizer = Arc::new(synchronizer);
+        for asset_type in asset_types {
+            let synchronizer = synchronizer.clone();
+            let rx = rx.resubscribe();
+
+            match asset_type {
+                AssetType::Fungible => {
+                    tokio::spawn(async move {
```

Comment:
Don't we need to add these tasks into JoinSet and then wait for them to complete?

Reply:
Fixed